sentry-android-replay/src/main/java/io/sentry/android/replay/screenshot/CanvasStrategy.kt

```diff
@@ -0,0 +1,890 @@
+import android.graphics.Bitmap
+import android.graphics.BlendMode
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.DrawFilter
+import android.graphics.Matrix
+import android.graphics.Mesh
+import android.graphics.NinePatch
+import android.graphics.Paint
+import android.graphics.Path
+import android.graphics.Picture
+import android.graphics.PixelFormat
+import android.graphics.PorterDuff
+import android.graphics.Rect
+import android.graphics.RectF
+import android.graphics.Region
+import android.graphics.RenderNode
+import android.graphics.fonts.Font
+import android.graphics.text.MeasuredText
+import android.media.Image
+import android.media.ImageReader
+import android.os.Handler
+import android.os.HandlerThread
+import android.view.View
+import io.sentry.SentryLevel
+import io.sentry.SentryOptions
+import io.sentry.android.replay.ScreenshotRecorderCallback
+import io.sentry.android.replay.ScreenshotRecorderConfig
+import io.sentry.android.replay.util.submitSafely
+import io.sentry.util.IntegrationUtils
+import java.util.LinkedList
+import java.util.WeakHashMap
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.LazyThreadSafetyMode.NONE
+import kotlin.math.roundToInt
+
+@SuppressLint("UseKtx")
+internal class CanvasStrategy(
+  private val executor: ScheduledExecutorService,
+  private val screenshotRecorderCallback: ScreenshotRecorderCallback?,
+  private val options: SentryOptions,
+  private val config: ScreenshotRecorderConfig,
+) : ScreenshotStrategy {
+
+  private var screenshot: Bitmap? = null
+  private val prescaledMatrix by
+    lazy(NONE) { Matrix().apply { preScale(config.scaleFactorX, config.scaleFactorY) } }
+  private val lastCaptureSuccessful = AtomicBoolean(false)
+  private val textIgnoringCanvas = TextIgnoringDelegateCanvas()
+  private var cachedPixels: IntArray? = null
+
+  private val freePictures: MutableList<PictureReaderHolder> =
+    LinkedList(
+      listOf(
+        PictureReaderHolder(config.recordingWidth, config.recordingHeight),
+        PictureReaderHolder(config.recordingWidth, config.recordingHeight),
+      )
+    )
+  private val unprocessedPictures: MutableList<PictureReaderHolder> = LinkedList()
+
+  private val processingThread = HandlerThread("screenshot_recorder.canvas")
+  private val processingHandler: Handler
+
+  init {
+    processingThread.start()
+    processingHandler = Handler(processingThread.looper)
+
+    IntegrationUtils.addIntegrationToSdkVersion("ReplayCanvasStrategy")
+  }
+
+  private val pictureRenderTask = Runnable {
+    val holder: PictureReaderHolder? =
+      synchronized(unprocessedPictures) {
+        when {
+          unprocessedPictures.isNotEmpty() -> unprocessedPictures.removeAt(0)
+          else -> null
+        }
+      }
+    if (holder == null) {
+      return@Runnable
+    }
+    try {
+      holder.reader.setOnImageAvailableListener(
```

Comment:
Is it possible to set this callback only once, or do we have to do it every time this Runnable runs?